### ISSUE TITLE
screenshot_on_failure = false is ignored on 0.3.2

### ISSUE THREAD

**Issue by pablodcar** [getgauge/gauge#309]
### Expected behavior
When `screenshot_on_failure = false ` on env/default/default.properties, screenshots must not be taken and included in reports. 

### Actual behavior
Screenshots are included in reports. It use to work in a previous version, Gauge 0.2.X but when I updated to 0.3.2 screenshots are always taken. 

### Steps to reproduce
1. Set `screenshot_on_failure = false ` on env/default/default.properties
2. Run an scenario that fails
3. See logs and report. If running in a headless environment, a line saying `Failed to take regular screenshot: headless environment` appears, on my local environment the screenshot is taken.

### Gauge version
```
Gauge version: 0.3.2

Plugins
-------
html-report (2.0.0)
java (0.3.4)
```

**Comment by mahendrakariya** [comment]
@pablodcar Thanks a lot for reporting the issue!
Could you please tell us if you are passing `--env` flag while running Gauge?

**Comment by kaustavdm** [comment]
Cannot reproduce.

**Comment by pablodcar** [comment]
Yes, I'm using the --env="qa" flag. On env/default/default.properties I have `screenshot_on_failure = false` while on env/qa/user.properties I did not redefine the `screenshot_on_failure` config.

**Comment by mahendrakariya** [comment]
Looks like this bug got introduced when we implemented #282.
We will be fixing it soon. In the meanwhile, as a workaround you can copy the `default.properties` file in the env directory that you are passing to the `--env` flag.
